config/providers/azure/sc-config.yaml

```diff
@@ -9,3 +9,8 @@ fluentd:
       # azcopy is used by az CLI for downloading files
       azureCopyBufferGB: 0.3
       azureCopyConcurrency: 8
+
+objectStorage:
+  sync:
+    ## If Harbor or Thanos are using swift then we will automatically use swift for the sync, regardless of the value set for type.
```

Comment:
I don't understand this comment. I'm sure it is obvious once you know why but to someone that does not it's not clear what is going on here.

Does it mean that the value of `objectStorage.sync.destinationType` is completely ignored if `.harbor.persistence.type=swift` or `.thanos.objectStorage.type=swift` or that the sync of those specific buckets are ignoring the value of `objectStorage.sync.destinationType`? I assume the latter?

To me the object storage configuration throughout the project feels a bit complicated. It really shows in the complex "set-me-if" cases. I wonder if we could simplify it somehow? Worthy of an issue?

Reply:
Time spent simplifying is time saved dealing with complexity in the future but I agree it's not high priority.

At least having tests for each of the config cases is something I'd like to se. I would be fine just seeing "config -> rendered result" and some "missing dependent config option / set-me-if-x" unit tests.

Reply:
Yeah I agree that it would be nice with both simplifications and some proper testing. I think you're saying that you'd like an issue for the testing, and I think that is reasonable.

I also reworded the comment in the config slightly, let me know if that is better.

Reply:
Issue for adding tests and validation. PTAL and feel free to update it if I missed something.
https://github.com/elastisys/compliantkubernetes-apps/issues/2320